invokeai/app/invocations/image.py

```diff
@@ -118,6 +124,80 @@ def invoke(self, context: InvocationContext) -> ImageOutput:
             image=image,
         )
 
+class DownloadImageInvocation(BaseInvocation):
+    """Fetches an image from a URL and provides it as output."""
+
+    type: Literal["download_image"] = "download_image"
+
+    # Inputs
+    image_url: str = Field(description="The URL of the image to download")
+
+    def invoke(self, context: InvocationContext) -> ImageOutput:
+        r = requests.get(self.image_url)
+        a = urlparse(self.image_url)
+
+        # TODO: how to handle failure?
+        if r.ok and r.headers["content-type"].startswith("image/"):
+            raw_image = io.BytesIO()
+            for chunk in r.iter_content(chunk_size=128):
+                raw_image.write(chunk)
+            raw_image.flush()
+            raw_image.seek(0)
+
+            return build_image_output(
+                image_type=ImageType.UPLOAD,
+                image_name=os.path.basename(a.path),
+                image=Image.Image.frombytes(raw_image)
+            )
+
+RESIZE_RESAMPLING_MODE = Literal[
+    "nearest", "box", "bilinear", "hamming", "bicubic", "lanczos"
+]
+
+class ResizeImageInvocation(BaseInvocation, PILInvocationConfig):
+    """Resizes an image to a specified size."""
+
+    type: Literal["resize"] = "resize"
+
+    # Inputs
+    image: ImageField = Field(default=None, description="The image to resize")
+    size: int = Field(default=512, gt=0, description="The size of the resized image's longest side")
+    resample_mode: Optional[RESIZE_RESAMPLING_MODE] = Field(default="bicubic", description="The resampling mode to use when resizing the image")
+
+    def invoke(self, context: InvocationContext) -> ImageOutput:
+        image = context.services.images.get(
+            self.image.image_type, self.image.image_name
+        )
+
+        resample_mode = Image.Resampling.BICUBIC
+        if self.resample_mode:
+            if self.resample_mode == "nearest":
+                resample_mode = Image.Resampling.NEAREST
+            elif self.resample_mode == "box":
+                resample_mode = Image.Resampling.BOX
+            elif self.resample_mode == "bilinear":
+                resample_mode = Image.Resampling.BILINEAR
+            elif self.resample_mode == "hamming":
+                resample_mode = Image.Resampling.HAMMING
+            elif self.resample_mode == "lanczos":
+                resample_mode = Image.Resampling.LANCZOS
+
+        # TODO: We should offer the PIL resampling options here, but how to represent that enum class?
+        resize_image = make_thumbnail(image, self.size, resampling_mode=resample_mode)
+
+        image_type = ImageType.INTERMEDIATE
```

Comment:
Pretty sure we want RESULT here. What if this node is the final in a chain of nodes?

Tbh I'm not sure what INTERMEDIATES are for, I guess this was intended for progress images but we've dropped support for accurate decoding those for the moment. @Kyle0654